### ISSUE TITLE
question about yolo-world default text features?

### ISSUE THREAD

**Issue by jiangguisyp** [ultralytics/ultralytics#15459]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

![K20240813-203351](https://github.com/user-attachments/assets/2ae8e802-a0bc-4105-9422-0f3804eef4ae)
In my understanding，the random initialized txt_feats are the clip features of COCO 80 classes, that is why we can use the yolo-world model directly without using set_classes method.  But why do the random values "equal" clip features?

**Comment by Y-T-G** [comment]
You're supposed to run `model.set_classes()` before prediction which sets the features according to your text.

**Comment by jiangguisyp** [comment]
> You're supposed to run `model.set_classes()` before prediction which sets the features according to your text.

I understand the workflow of set_classes, which calls encode method of CLIP to compute text feature;

what bothers me is the default workflow without using set_classes, which detects COCO 80 classes, where does the code compute or load corresponding text features?

**Comment by Y-T-G** [comment]
The `txt_feats` of the original model is part of the checkpoint. When the weights are loaded, it also loads the `txt_feats` which is by default for COCO classes.

**Comment by Y-T-G** [comment]
```python
import torch

model = torch.load("yolov8s-world.pt")
print(model["model"].txt_feats)
```

**Comment by jiangguisyp** [comment]
> ```python
> import torch
> 
> model = torch.load("yolov8s-world.pt")
> print(model["model"].txt_feats)
> ```

@Y-T-G Thanks for your reply, but the problem is still bothers me.

I check the model["model"].txt_feats, the values are exactly equal to the txt_feats initialized by random in method __init__  as shown in the figure below

![K20240813-203351](https://github.com/user-attachments/assets/837f5413-3e02-41c2-835a-1d3f6c799fb5)

I want to know how this is done ？

**Comment by Y-T-G** [comment]
No, they are not random values. I don't know how you checked because they're not the same.

```python
In [1]: from ultralytics import YOLOWorld

In [2]: model = YOLOWorld("yolov8s-world.pt")

In [3]: model.model.txt_feats
Out[3]:
tensor([[[ 1.5075e-02,  7.6512e-03,  1.3623e-02,  ..., -2.3560e-02, -1.9117e-02,  1.3763e-03],
         [ 9.1134e-03, -1.3279e-02,  1.2939e-02,  ...,  1.2770e-02,  2.2071e-02, -2.3817e-02],
         [ 6.0259e-03,  9.7333e-03,  3.5352e-03,  ..., -2.2382e-02, -2.4746e-02,  1.0388e-02],
         ...,
         [ 7.5728e-05, -2.9587e-02, -3.6387e-03,  ..., -2.5101e-02, -4.5286e-02, -2.0711e-02],
         [-6.4858e-03,  5.8284e-03, -3.5350e-02,  ..., -1.6866e-02, -3.4616e-02, -7.0484e-03],
         [-2.2845e-02,  1.5922e-02, -3.1341e-02,  ..., -9.7644e-03, -4.8756e-02,  2.7139e-02]]])

In [4]: model = YOLOWorld("yolov8s-world.yaml")

In [5]: model.model.txt_feats
Out[5]:
tensor([[[ 2.5870,  0.2179,  0.8794,  ...,  0.2834,  0.3971,  0.4016],
         [-0.5037, -1.6359, -0.3010,  ...,  0.9789, -1.3331, -0.8981],
         [-0.9858, -0.6494,  0.0452,  ..., -0.1674, -0.0262,  2.0687],
         ...,
         [-1.2671,  1.6959,  0.8101,  ..., -0.8029, -0.9388, -0.3166],
         [ 0.7266, -0.6775, -0.1588,  ..., -0.6841,  0.7802,  0.8841],
         [ 0.2127,  0.4715,  0.1776,  ...,  1.8764,  0.4042, -0.8198]]])
```

**Comment by jiangguisyp** [comment]
> No, they are not random values if load from checkpoint. I don't know how you checked because they're not the same.
> 
> ```python
> In [1]: from ultralytics import YOLOWorld
> 
> In [2]: model = YOLOWorld("yolov8s-world.pt")
> 
> In [3]: model.model.txt_feats
> Out[3]:
> tensor([[[ 1.5075e-02,  7.6512e-03,  1.3623e-02,  ..., -2.3560e-02, -1.9117e-02,  1.3763e-03],
>          [ 9.1134e-03, -1.3279e-02,  1.2939e-02,  ...,  1.2770e-02,  2.2071e-02, -2.3817e-02],
>          [ 6.0259e-03,  9.7333e-03,  3.5352e-03,  ..., -2.2382e-02, -2.4746e-02,  1.0388e-02],
>          ...,
>          [ 7.5728e-05, -2.9587e-02, -3.6387e-03,  ..., -2.5101e-02, -4.5286e-02, -2.0711e-02],
>          [-6.4858e-03,  5.8284e-03, -3.5350e-02,  ..., -1.6866e-02, -3.4616e-02, -7.0484e-03],
>          [-2.2845e-02,  1.5922e-02, -3.1341e-02,  ..., -9.7644e-03, -4.8756e-02,  2.7139e-02]]])
> 
> In [4]: model = YOLOWorld("yolov8s-world.yaml")
> 
> In [5]: model.model.txt_feats
> Out[5]:
> tensor([[[ 2.5870,  0.2179,  0.8794,  ...,  0.2834,  0.3971,  0.4016],
>          [-0.5037, -1.6359, -0.3010,  ...,  0.9789, -1.3331, -0.8981],
>          [-0.9858, -0.6494,  0.0452,  ..., -0.1674, -0.0262,  2.0687],
>          ...,
>          [-1.2671,  1.6959,  0.8101,  ..., -0.8029, -0.9388, -0.3166],
>          [ 0.7266, -0.6775, -0.1588,  ..., -0.6841,  0.7802,  0.8841],
>          [ 0.2127,  0.4715,  0.1776,  ...,  1.8764,  0.4042, -0.8198]]])
> ```

@Y-T-G I figured it out, thanks for your patient guidance！
